docs/rst/tutorials/core/deployment/ds_wan_tcp/ds_wan_tcp.rst

```diff
@@ -0,0 +1,369 @@
+  This container's default gateway is redirected to the discovery *server*, which would behave as a router too.
+
+* ``router``: the container is included in both ``listener_net`` and ``wan_net`` networks.
+  The IP addresses has been manually set to ``10.2.1.1`` in the ``listener_net``, and ``10.3.2.1`` in the ``wan_net``.
+  This container's default gateway is redirected to the discovery *server*.
+  The ``iptables`` has been configured to redirect traffic from any network and interface.
+
+
+Please, include the following ``compose.yml`` file in the workspace.
+
+.. literalinclude:: /resources/tutorials/core/deployment/ds_wan_tcp_tutorial/compose.yml
+    :language: yaml
+
+Run the example
+---------------
+
+Make sure that the workspace has been set with all the previous files, and the docker networks and ``iptables`` have been set too.
+
+.. code-block:: bash
+
+    ├ <workspace>
+    ·  ├ compose.yml
+    ·  ├ Dockerfile
+    ·  ├ listener_configuration.xml
+    ·  ├ server_configuration.xml
+    ·  └ talker_configuration.xml
+
+As explained in the :ref:`tutorials_deployment_ds_wan_tcp_overview`, the expected behavior is that both ``talker`` and ``listener`` are able to connect to the discovery *server*, discover each other, and send and receive (respectively) the *HelloWorld* example messages over the WAN simulation using TCP.
+
+Run the example:
+
+.. code-block:: bash
+
+    cd <workspace>
+    docker compose -f compose.yml up --build
+
+Clean workspace
+---------------
+
+Stop the example by pressing ``Ctrl + C``, and stop the containers by running:
+
+.. code-block:: bash
+
+    docker compose -f compose.yml down
+
+The docker networks, containers and images can be removed using the docker ``prune`` argument, but using the ``rm`` argument plus the identifiers would remove only the ones created for this tutorial:
+
+.. code-block:: bash
+
+    docker network rm listener_net talker_net wan_net
+    docker container rm <workspace_name>_fast_dds_discovery_server_1 <workspace_name>_router_1 <workspace_name>_ros_listener_1 <workspace_name>_ros_talker_1
+    docker image rm <workspace_name>_fast_dds_discovery_server:latest <workspace_name>_router:latest <workspace_name>_ros_listener:latest <workspace_name>_ros_talker:latest
+
+.. note::
+
+    The ``iptables`` configuration would be removed automatically each time the system gets rebooted.
+
+TCP over real WAN with Discovery Server
+---------------------------------------
+
+The configuration of the docker networks and the ``iptables`` has been performed to simulate the WAN scenario locally.
+The requirements to achieve TCP communication over WAN with Discovery Server as EDP in a real deployment are launching the three elements of the communication (``talker``, ``listener`` and discovery *server*) with their corresponding XML configurations applied, and setting the proper firewall or router configuration rules.
```

Comment:
I think that the Discovery Server acts as an entire discovery mechanism (including both PDP and EDP phases)